domain/src/Aggregates/SharePooling/Services/QuantityAdjuster/QuantityAdjuster.php

```diff
@@ -22,7 +23,13 @@ public static function revertDisposal(
         SharePoolingTransactions $transactions,
     ): void {
         foreach (self::getAcquisitions($disposal->sameDayQuantityBreakdown, $transactions) as $acquisition) {
-            $acquisition->decreaseSameDayQuantity($disposal->sameDayQuantityBreakdown->quantityMatchedWith($acquisition));
+            try {
+                $acquisition->decreaseSameDayQuantity($disposal->sameDayQuantityBreakdown->quantityMatchedWith($acquisition));
+            } catch (SharePoolingTokenAcquisitionException) {
+                // @TODO When re-acquiring within 30 days an asset that was disposed of on the same day it was acquired,
+                // decreasing the same-day quantity of the concerned acquisitions fails, because at the time the latter
+                // were recorded within the SharePoolingTokenAcquired events that had no same-day quantity yet
+            }
```

Comment:
ℹ️ This is really not a fix – just a plaster for now. Note that the impact is minimal at the moment, as decreasing this quantity would basically reset the acquisition's quantities, and when this bug occurs currently, the acquisition is effectively in a reset state already.

There may be more complex scenarios in the future though, in which case this could be an issue.